src/integration.test.ts

```diff
@@ -24,7 +24,9 @@ describe("Integration Test", (): void => {
   let util: typeof import("./util.js");
   let docker: typeof import("./docker.js");
 
-  beforeAll(async (): Promise<void> => {
+  beforeEach(async (): Promise<void> => {
+    inMemoryCache = {};
+    state = {};
```

Comment:
Nit: recommend either moving these assignments below the imports with a blank line between (on the basis that imports tend to be at the top of files, so they could at least be at the top of the function in this case) or performing the assignments in the same order as the declarations (on the basis of convention).